protocol-units/settlement/mcr/contracts/src/staking/MovementStaking.sol

```diff
@@ -141,9 +129,25 @@ contract MovementStaking is
         uint256 epoch,
         address custodian,
         address attester,
+        address delegator,
         uint256 amount
     ) internal {
-        epochStakesByDomain[domain][epoch][custodian][attester] -= amount;
+        epochStakesByDomain[domain][epoch][custodian][attester][
+            delegator
+        ] -= amount;
+
+        if (
+            epochStakesByDomain[domain][epoch][custodian][attester][
+                delegator
+            ] == 0
+        ) {
```

Comment:
Here as well: the value has just been decremented, could we hold on to the reference and get the value instead of spending gas on 5 nested lookups another time?

Reply:
Yep, good call.